src/main/java/patching/PatchManager.java

```diff
@@ -124,13 +153,16 @@ public void run() {
                 response -> {
                     if (response.getStatus() == 200) {
                         synchronized (batchingCtrl.patchBatchingQueue) {
-                            logger.debug(String.format("PatchManager: Removing patches %s; patch queue is currently %s", batchingCtrl.patchBatchingQueue.subList(0, patches.length), batchingCtrl.patchBatchingQueue).replace("\n", "\\n") + "\n");
-                            logger.debug(String.format("PatchManager: Removing patches %s; patch done queue is currently %s", batchingCtrl.patchBatchingQueue.subList(0, patches.length), batchingCtrl.patchDoneQueue).replace("\n", "\\n") + "\n");
+                            logger.debug(String.format("PatchManager: Removing patches %s; patch queue is currently %s", batchingCtrl.patchBatchingQueue.subList(0, patches.length), batchingCtrl.patchBatchingQueue).replace("\n", "\\n"));
+                            logger.debug(String.format("PatchManager: Removing patches %s; patch done queue is currently %s", batchingCtrl.patchBatchingQueue.subList(0, patches.length), batchingCtrl.patchDoneQueue).replace("\n", "\\n"));
                             // Remove the sent patches
                             for (int i = 0; i < patches.length; i++) {
-                                batchingCtrl.patchDoneQueue.add(batchingCtrl.patchBatchingQueue.remove(0));
+                                Patch p = batchingCtrl.patchBatchingQueue.remove(0);
```

Comment:
There's actually no way for us to easily identify which patch was accepted and which failed. As of right now, the server accepts all or nothing, so this assumption is safe. We need to send it back so we can properly transform any old notifications.